caffe2/operators/conv_dgrad_op_cudnn.cc

```diff
@@ -0,0 +1,270 @@
+#include "caffe2/operators/conv_op_cudnn_base.h"
+
+namespace caffe2 {
+
+static constexpr size_t kNUM_CUDNN_BWD_DATA_ALGS = 6;
+
+class CudnnConvDataGradientOp final : public CudnnConvOpBase {
+ public:
+  CudnnConvDataGradientOp(const OperatorDef& operator_def, Workspace* ws)
+      : CudnnConvOpBase(operator_def, ws) {}
+  ~CudnnConvDataGradientOp() {}
+
+  template <typename T_X, typename T_W, typename T_DY, typename MATH, typename T_DX>
+  bool DoRunWithType();
+
+  bool RunOnDevice() override;
+
+ private:
+  cudnnConvolutionBwdDataAlgo_t algo_;
+  AlgorithmsCache<cudnnConvolutionBwdDataAlgo_t> algo_cache_;
+
+  // in/out: {X, W, dY} -> {dX}
+  INPUT_TAGS(INPUT, FILTER, OUTPUT_GRAD);
+  OUTPUT_TAGS(INPUT_GRAD);
+};
+
+
+template <typename T_X, typename T_W, typename T_DY,
+          typename MATH,
+          typename T_DX>
+bool CudnnConvDataGradientOp::DoRunWithType() {
+  auto& X = Input(INPUT);
+  auto& filter = Input(FILTER);
+  auto& dY = Input(OUTPUT_GRAD);
+  auto* dX = Output(INPUT_GRAD);
+
```

Comment:
Yes, lets do that please.
The reason I asked for this is to remove the redefinition of dX in line 160 because it doesn't compile with -Werror=shadow-compatible-local and to remove resizing dX for every group in 216.